src/utils/details.cpp

```diff
@@ -29,6 +29,38 @@
 namespace mrcpp {
 namespace details {
 
+// helper function: parse a string and returns the nth integer number
+int get_val(char *line, int n) {
+    char *p = line;
+    int len = 0;
+    for (int i = 0; i < n - 1; i++) {
+        // jump over n-1 first numbers
+        while (*p < '0' || *p > '9') p++;
+        while (*p >= '0' && *p <= '9') p++;
+    }
+    while (*p < '0' || *p > '9') p++;
+    char *s = p;
+    while (*s >= '0' && *s <= '9') {
+        s++;
+        line[len] = p[len];
+        len++;
+    }
+    p[len] = 0;
+    return atoi(p);
+}
+
+/** @brief returns the current memory usage of this process, in kB */
+int get_memory_usage() {
+    FILE *file = fopen("/proc/self/statm", "r");
```

Comment:
Does this work on macOS?

Reply:
Good question. @gitpeterwind ?

Reply:
I haven't a Mac to try. (But I remember that the /proc/self/statm "directory" is special; it is not a regular directory in the sense that it is located on a disk.) . Otherwise, do you have any better measure of total memory usage? It is a very necessary diagnostic when developing/testing, so even if it cannot be used on Mac, I would prefer keep it.

Reply:
I will add a back door returning -1 if the "file" doesn't exist

Reply:
Yes, it should be kept. I don't know other methods to measure total memory consumption. The alternative of modifying `malloc`/`free` or writing custom allocators is too time consuming to implement.